### PR TITLE
Fixes/renaming related bugs

### PR DESCRIPTION
**Pull request by caneruguz**
## Purpose

Improve renaming on the Project organizer. Fixes three issues:
1. Renaming did not refresh the toolbar name
2. Items whose parent were not folders did not show rename div (broken due to a previous change I made)
3. Renaming did not refresh every instance of that item

https://trello.com/c/7ghezzfW

## Changes

1. Refresh didn't happen because folder was reloading. Moved the click trigger to when lazyloading finishes on that item. 
2. Added the div back to the section for components and links
3. Instead of a folder refresh, renaming does a full refresh

## Side effects
Reloading entire dashboard may have consequences but we save open state so that shouldn't be a problem. There isn't a major observable delay.

### CODE REVIEW

**Comment by brianjgeiger** [comment]
As far as side effects of the third item, I would only expect that it will take longer to deal with. The original Project Organizer doesn't handle that case, and the testing card created for it expects that a project renamed will not reflect those changes everywhere until reloading of the page or the folder holding the project.

**Comment by sloria** [comment]
@caneruguz This seems to have caused a different regression: 

On Firefox (35.0), When I click "Rename" the rename is successful, but the toolbar does not return to "normal" mode (it stays in "edit" mode).

File: website/static/js/projectorganizer.js [hunk 1]
```diff
@@ -697,8 +701,14 @@ function expandStateLoad(item) {
             if (item.children[i].data.expand) {
                 tb.updateFolder(null, item.children[i]);
             }
+            console.log(tb.multiselected[0].data.node_id, item.children[i].data.node_id);
```
Comment (sloria) [inline website/static/js/projectorganizer.js]: Do not check in `console.log` statements.

File: website/static/js/projectorganizer.js [hunk 2]
```diff
@@ -417,8 +421,8 @@ function _showProjectDetails(event, item, col) {
                 };
             postAction = $osf.postJSON(url, postData);
             postAction.done(function () {
-                treebeard.updateFolder(null, theParentNode);
-                triggerClickOnItem.call(treebeard, item);    
+                treebeard.updateFolder(null, treebeard.find(1));
```
Comment (sloria) [inline website/static/js/projectorganizer.js]: Does this update the root node? If so, why is this necessary?

**Comment by sloria** [comment]
1st pass finished. :lollipop:

File: website/static/js/projectorganizer.js (hunk 2 above)
Comment (caneruguz) [inline website/static/js/projectorganizer.js]: It's explained in the description, because if the element has multiple locations (for instance projects put to two different folders that are open) changing name of one has to change the name of the other. Reloading the folder completely is the most efficient way to achieve this I think. Otherwise there will need to be a full lookup. 

Brian's note is important here too. What is the suggestion of the team? I can update the code.

Comment (brianjgeiger) [inline website/static/js/projectorganizer.js]: If you are updating the root, which I don't recommend, you don't need to update the parent node.

**Comment by caneruguz** [comment]
It looks like in Firefox rename never returns to normal state. I fixed it in this pull request
https://github.com/CenterForOpenScience/osf.io/pull/1548

File: website/static/js/projectorganizer.js (hunk 2 above)
Comment (sloria) [inline website/static/js/projectorganizer.js]: I see. Sorry, I missed that part in the description. 

@brianjgeiger Did you intentionally not update the whole grid on rename?

Comment (sloria) [inline website/static/js/projectorganizer.js]: I think refreshing the root node after a rename makes sense for now.

Comment (brianjgeiger) [inline website/static/js/projectorganizer.js]: Yes, I intentionally did not. It's potentially fairly slow for such a simple action.

Comment (sloria) [inline website/static/js/projectorganizer.js]: Keeping it for now. We'll make sure to test on large numbers of projects.
